src/users/userHelpers.js

```diff
@@ -37,13 +37,43 @@ const statusOrder = (presence: UserPresence): number => {
   }
 };
 
-export const sortUserList = (users: User[], presences: PresenceState): User[] =>
-  [...users].sort(
-    (x1, x2) =>
-      statusOrder(presences[x1.email]) - statusOrder(presences[x2.email])
-      || x1.full_name.toLowerCase().localeCompare(x2.full_name.toLowerCase()),
+export const sortAlphabetically = (users: User[]): User[] =>
+  [...users].sort((x1, x2) => x1.full_name.toLowerCase().localeCompare(x2.full_name.toLowerCase()));
+
+export const filterUserStartWith = (users: User[], filter: string = ''): User[] =>
+  users.filter(user => user.full_name.toLowerCase().startsWith(filter.toLowerCase()));
+
+export const filterUserEmailStartWith = (users: User[], filter: string = ''): User[] =>
+  users.filter(user => user.email.toLowerCase().startsWith(filter.toLowerCase()));
+
+export const filterUserByInitials = (users: User[], filter: string = ''): User[] =>
+  users.filter(user =>
+    user.full_name
+      .replace(/(\s|[a-z])/g, '')
+      .toLowerCase()
+      .startsWith(filter.toLowerCase()),
   );
 
+export const getUniqueUsers = (users: User[]): User[] => uniqby(users, 'email');
+
+export const sortByPositionOfKeyword = (users: User[], filter: string = ''): User[] => {
+  if (filter === '') {
+    return users;
+  }
+  const beginOfUserName = filterUserStartWith(users, filter);
+  const beginOfInitialName = filterUserByInitials(users, filter);
+  const beginOfEmail = filterUserEmailStartWith(users, filter);
+  return getUniqueUsers([...beginOfUserName, ...beginOfEmail, ...beginOfInitialName, ...users]);
+};
+
+export const sortUserList = (
+  users: User[],
+  presences: PresenceState,
+  filter: string = '',
+): User[] =>
+  [...sortByPositionOfKeyword(users, filter)].sort(
+    (x1, x2) => statusOrder(presences[x1.email]) - statusOrder(presences[x2.email]),
+  );
```

Comment:
Also, please make sure there's ordinarily a blank line between successive functions, for readability's sake.